Git-commit: d65f69deac5cf916cc27762bccde8c8874421cae
From: Heiko Carstens <hca@linux.ibm.com>
Date: Thu, 10 Dec 2020 22:30:09 +0100
Subject: [PATCH] s390/irq: select HAVE_IRQ_EXIT_ON_IRQ_STACK
References: bsn#19
Patch-mainline: v5.10.50

[ Upstream commit 9ceed9988a8e6a1656ed2bdaa30501cf0f3dd925 ]

irq_exit() is always called on async stack. Therefore select
HAVE_IRQ_EXIT_ON_IRQ_STACK and get a tiny optimization in
invoke_softirq().

Signed-off-by: Heiko Carstens <hca@linux.ibm.com>
Signed-off-by: Sasha Levin <sashal@kernel.org>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 arch/s390/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/s390/Kconfig b/arch/s390/Kconfig
index 4a2a12be04c9..dc5c3e6fd200 100644
--- a/arch/s390/Kconfig
+++ b/arch/s390/Kconfig
@@ -154,6 +154,7 @@ config S390
 	select HAVE_FUTEX_CMPXCHG if FUTEX
 	select HAVE_GCC_PLUGINS
 	select HAVE_GENERIC_VDSO
+	select HAVE_IRQ_EXIT_ON_IRQ_STACK
 	select HAVE_KERNEL_BZIP2
 	select HAVE_KERNEL_GZIP
 	select HAVE_KERNEL_LZ4
-- 
2.31.1

